### PR TITLE
Fix script calculations

### PR DESCRIPTION
**Pull request by erwinmombay**
if AMP_CONFIG.esm is truthy, file extension that is dynamically inserted should be an `mjs` file as mismatch of files could cause unexpected errors (they are not of the same build, modern syntax clashing with old style transpilations, etc)

### CODE REVIEW

File: test/unit/test-extension-location.js [hunk 1]
```diff
@@ -203,3 +204,192 @@ describes.sandboxed('Extension Location', {}, () => {
     });
   });
 });
+
+describes.sandboxed('Module Extension Location', {}, () => {
```
Comment (samouri) [inline test/unit/test-extension-location.js]: Nit: I may be missing something, but this number of tests feels like overkill to me.
I think we'd still get good coverage with just one or two tests within the `describe('get correct script source')` grouping.

**Comment by erwinmombay** [comment]
@samouri i'll be doing a slow rollout pretty soon (this week rolling over to next) and monitor it

File: test/unit/test-extension-location.js (hunk 1 above)
Comment (erwinmombay) [reply test/unit/test-extension-location.js]: agree with you for sure, was just in a rush and duplicated the tests

**Comment by erwinmombay** [comment]
going to force merge this as the integration tests are just flakey and I need this for a patch

**Pull request merged**